src/client/app/redux/componentHooks.ts

```diff
@@ -0,0 +1,38 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { selectInitComplete, selectSelectedLanguage } from './slices/appStateSlice';
+import { selectCurrentUserRole, selectIsAdmin } from './slices/currentUserSlice';
+import { useAppSelector } from './reduxHooks';
+import localeData, { LocaleDataKey } from '../translations/data';
+import { createIntlCache, createIntl, defineMessages } from 'react-intl';
+
+export const useWaitForInit = () => {
+	const isAdmin = useAppSelector(selectIsAdmin);
+	const userRole = useAppSelector(selectCurrentUserRole);
+	const initComplete = useAppSelector(selectInitComplete);
+	return { isAdmin, userRole, initComplete }
+}
+
+// Overloads to support TS key completions
+type TranslateFunction = {
+	(messageID: LocaleDataKey): string;
+	(messageID: string): string;
+}
+
+// usage
+// const translate = useTranslate()
+// translate('myKey')
+export const useTranslate = () => {
```

Comment:
I'm trying to figure out how this relates to src/client/app/utils/translate.ts and the fact that some places use one or the other.

Reply:
Need to resolve store more generally so will address in future.